client/app/license/spla/add/license-spla-add.controller.js

```diff
@@ -1,53 +1,66 @@
-angular.module("Module.license").controller("LicenseSplaAddCtrl", ($scope, $translate, License, Alerter) => {
-    "use strict";
+angular.module("Module.license").controller("LicenseSplaAddCtrl", class LicenseSplaAddCtrl {
 
-    $scope.model = {
-        options: {
+    constructor ($state, $translate, Alerter, License) {
+        this.$state = $state;
+        this.$translate = $translate;
+        this.Alerter = Alerter;
+        this.License = License;
+    }
+
+    $onInit () {
+        this.options = {
             availableServers: null,
             availableTypes: null
-        },
-        selected: {
+        };
+
+        this.selected = {
             serial: null,
             type: null,
             server: null
-        }
-    };
+        };
+    }
 
-    $scope.load = () => {
-        License.splaAddAvailableServers().then(
-            (data) => {
-                $scope.model.options = data;
+    addSpla () {
+        this.License.splaAdd(this.selected.server, { serialNumber: this.selected.serial, type: this.selected.type }).then(
+            () => {
+                this.Alerter.alertFromSWS(this.$translate.instant("license_spla_add_success"), true);
             },
-            (data) => {
-                $scope.resetAction();
-                Alerter.alertFromSWS($translate.instant("license_spla_add_step1_loading_error"), data.data);
+            ({ data }) => {
+                this.Alerter.alertFromSWS(this.$translate.instant("license_spla_add_fail"), data);
             }
         );
-    };
+    }
 
-    $scope.addSpla = () => {
-        $scope.resetAction();
-        License.splaAdd($scope.model.selected.server, { serialNumber: $scope.model.selected.serial, type: $scope.model.selected.type }).then(
-            () => {
-                Alerter.alertFromSWS($translate.instant("license_spla_add_success"), true);
-            },
+    close () {
```

Comment:
Hey,

What do you close here? A modal?

If you do that for the modal, it should be managed by the modal itself, not by every controller which needs a modal?

Else, you are going to get these case:

controller1 close() $state.go("^")
controller2 close() $state.go("^")
controller3 close() $state.go("^")

x100? :D

Ok ok, what do you think about this? :)

Reply:
@jleveugle ok for this, I add a method, "endAction" who close the modal, but since it is not only used to close modal without any context (endAction is used after final order action), is it ok for you ?